SSA/Experimental/Bits/Fast/BitStream.lean

```diff
@@ -363,10 +369,78 @@ Crucially, our decision procedure works by considering which equalities hold for
 --     (∀ w, (x w + y w) = z w) ↔ (∀ w, (ofBitVec (x w)) + (ofBitVec (y w)) ) := by
 --   have ⟨h₁, h₂⟩ : True ∧ True := sorry
 --   sorry
-@[simp] theorem ofBitVec_sub : ofBitVec (x - y) = (ofBitVec x) - (ofBitVec y) := sorry
-@[simp] theorem ofBitVec_add : ofBitVec (x + y) = (ofBitVec x) + (ofBitVec y) := sorry
-@[simp] theorem ofBitVec_neg : ofBitVec (-x) = -(ofBitVec x) := sorry
-@[simp] theorem ofBitVec_not : ofBitVec (~~~ x) = ~~~ (ofBitVec x) := sorry
+
+variable {w : Nat} {x y : BitVec w} {a b a' b' : BitStream}
+
+local notation a " ≈ʷ  " b  => EqualUpTo w a b
+
+-- TODO: These sorries are difficult, and will be proven in a later Pull Request.
+@[simp] theorem ofBitVec_sub : ofBitVec (x - y) ≈ʷ (ofBitVec x) - (ofBitVec y)  := by
+  sorry
+
+@[simp] theorem ofBitVec_add : ofBitVec (x + y) ≈ʷ (ofBitVec x) + (ofBitVec y)  := by
+  sorry
+
+@[simp] theorem ofBitVec_neg : ofBitVec (- x) ≈ʷ  - (ofBitVec x) := by
+  sorry
+
+theorem equal_up_to_refl : a ≈ʷ a := by
+  intros  j _
+  rfl
+
+theorem equal_up_to_symm (e : a ≈ʷ b) : b ≈ʷ a := by
+  intros j h
+  symm
+  exact e j h
+
+theorem equal_up_to_trans (e1 : a ≈ʷ b) (e2 : b ≈ʷ c) : a ≈ʷ c := by
```

Comment:
Often with trans theorems its nice to implement `Trans` as it means `calc` can be used it proofs making things quite clean